packages/create-cloudflare/src/cli.ts

```diff
@@ -30,6 +32,39 @@ const WRANGLER_DEFAULTS = {
 export const main = async (argv: string[]) => {
 	const args = await parseArgs(argv);
 
+	// Print a newline
+	logRaw("");
+
+	if (args.autoUpdate && (await isUpdateAvailable())) {
+		await runLatest();
+	} else {
+		await runCli(args);
+	}
+};
+
+const isUpdateAvailable = async () => {
+	if (process.env.VITEST) {
+		return false;
+	}
+
+	// Use a spinner when running this check since it may take some time
+	const s = spinner(spinnerFrames.vertical);
+	s.start("Checking for newer version");
+	const latestVersion = await runCommand(
+		`npm info create-cloudflare@latest dist-tags.latest`,
+		{ silent: true, useSpinner: false }
+	);
+	s.stop();
+
+	return semver.gt(latestVersion, version);
+};
+
+export const runLatest = async () => {
+	const args = process.argv.slice(2);
+	await runCommand(`npm create cloudflare@latest ${args.join(" ")}`);
```

Comment:
shouldn't this use the current package manager instead of hardcoding `npm`?

Reply:
I think so too.

Reply:
Good catch!